python/ray/serve/batching.py

```diff
@@ -114,11 +116,15 @@ async def wait_for_batch(self) -> List[Any]:
         batch = []
         batch.append(await self.queue.get())
 
+        # Cache current max_batch_size and batch_wait_timeout_s for this batch
+        max_batch_size = self.max_batch_size
+        batch_wait_timeout_s = self.batch_wait_timeout_s
+
```

Comment:
```suggestion
        # Cache current max_batch_size and batch_wait_timeout_s for this batch.
        max_batch_size = self.max_batch_size
        batch_wait_timeout_s = self.batch_wait_timeout_s

```

Reply:
so the semantics here are that the settings will be picked up after the current batch finishes? sounds fine to me

Reply:
Yep, that's right.